documentation/modules/ROOT/pages/configuration/storage.adoc

```diff
@@ -0,0 +1,705 @@
+
+|[[schema-history-internal-kafka-query-timeout-ms]]<<schema-history-internal-kafka-query-timeout-ms,`schema.history.internal.kafka.query.timeout.ms`>>
+|3
+|The number of milliseconds to wait while fetching cluster information using Kafka admin client.
+
+|[[schema-history-internal-kafka-create-timeout-ms]]<<schema-history-internal-kafka-create-timeout-ms,`schema.history.internal.kafka.create.timeout.ms`>>
+|30
+|The number of milliseconds to wait while create kafka history topic using Kafka admin client.
+
+|[[schema-history-internal-producer]]<<schema-history-internal-producer,`schema.history.internal.producer.*`>>
+|No Default
+|Pass-through properties prefix for configuring how producer client interact with schema history topics.
+
+|[[schema-history-internal-consumer]]<<schema-history-internal-consumer,`schema.history.internal.consumer.*`>>
+|No Default
+|Pass-through properties prefix for configuring how consumer client interact with schema history topics.
+
+|===
+
+
+== File
+
+It's possible to persist the position (offsets) of the connector in a local file on disk. This allows Debezium to resume from where it left off after a restart. It's simple, fast, and ideal for single-node applications or testing scenarios.
+
+=== Offset Store
+
+[cols="35%a,10%a,55%a",options="header"]
+|===
+|Property
+|Default
+|Description
+
+|[[offset-storage-file]]<<offset-storage-file,`offset.storage`>>
+|No default
+|Must be set to `org.apache.kafka.connect.storage.FileOffsetBackingStore`
+
+|[[offset-storage-file-filename]]<<offset-storage-file-filename,`offset.storage.file.filename`>>
+|No default
+|Path to the file where Debezium stores source offsets.
+
+|[[offset-flush-interval-ms]]<<offset-flush-interval-ms,`offset.flush.interval.ms`>>
+|6000ms
+|Interval in milliseconds to flush the current offset state to the file.
+
+|===
+
+=== Internal schema history store
+
+[cols="35%a,10%a,55%a",options="header"]
+|===
+|Property
+|Default
+|Description
+
+|[[schema-history-internal-file]]<<schema-history-internal-file,`schema.history.internal`>>
+|No default
+|Must be set to `io.debezium.storage.file.history.FileSchemaHistory`
+|[[schema-history-internal-file-file]]<<schema-history-internal-file-file,`schema.history.internal.file`>>
+|No default
+|The path to the file that will be used to record the database schema history`
+
+|===
+
+== Memory
+
+`MemoryOffsetBackingStore` is a volatile, in-memory store used by Debezium Embedded to track source offsets. It keeps the offset state only during the application's runtime and loses it on shutdown or crash. It's ideal for testing or short-lived tasks but not suitable for production where offset persistence is needed.
+
+=== Offset Store
+
+[cols="35%a,10%a,55%a",options="header"]
+|===
+|Property
+|Default
+|Description
+
+|[[offset-storage-memory]]<<offset-storage-memory,`offset.storage`>>
+|No default
+|Must be set to `org.apache.kafka.connect.storage.MemoryOffsetBackingStore`
+
```

Comment:
If this and similar properties have required values, can they be coded to carry the appropriate defaults?

Reply:
Actually I don't know what could be the right default. Maybe the `MemoryOffsetBackingStore` but actually the code doesn't have a default value. WDYT @jpechane?

Reply:
No, they can't. This is exactly the way how you choose which storage do you use.